openqaoa/optimizers/result.py

```diff
@@ -251,6 +255,97 @@ def plot_probabilities(self, n_states_to_keep = None, figsize = (10,8),label='Pr
+            The linestyle for each parameter. Defaults to '--' for all parameters. 
+            If it is a string all parameters will use it, if it a list of strings the linestyle of each parameter will depend on one string of the list.
+        color: `list[str]` or `str`
+            The color for each parameter. Defaults to None for all parameters (matplotlib will choose the colors). 
+            If only one parameter is plot the color can be a string, otherwise it must be a list of strings.
+        ax: 'matplotlib.axes._subplots.AxesSubplot'
+            Axis on which to plot the graph. If none is given, a new figure will be created.
+
+        """
+
+        if ax is None:
+            ax = plt.subplots(figsize=figsize)[1]
+
+        ## creating a list of parameters to plot
+        # if param_to_plot is not given, plot all the parameters 
+        if param_to_plot is None: param_to_plot = list(range(len(self.n_shots[0]))) 
+        # if param_to_plot is a single value, convert to list
+        elif type(param_to_plot) == int: param_to_plot = [param_to_plot]
+        # if param_to_plot is not a list, raise error
+        if type(param_to_plot) != list: raise ValueError('`param_to_plot` must be a list of integers or a single integer')
+        else: 
+            for param in param_to_plot:
+                assert param < len(self.n_shots[0]) , f'`param_to_plot` must be a list of integers between 0 and {len(self.n_shots[0]) - 1}'
+
+        # if label is not given, create a list of labels for each parameter (only if there is more than 1 parameter)
+        if len(self.n_shots[0]) > 1: label = [f'Parameter {i}' for i in param_to_plot] if label is None else label
+        else: label = ['n. shots per parameter']
+
+        # if only one parameter is plotted, convert label and color to list if they are string
+        if len(param_to_plot) == 1:
+            if type(label) == str: label = [label]
+            if type(color) == str: color = [color]
+
+        # if param_top_plot is a list and label or color are not lists, raise error
+        if (type(label) != list) or (type(color) != list and color != None):
+            raise TypeError('`label` and `color` must be list of str')
+        # if label is a list, check that all the elements are strings
+        for l in label: assert type(l) == str, '`label` must be a list of strings'
+        # if color is a list, check that all the elements are strings
+        if color != None: 
+            for c in color: assert type(c) == str, '`color` must be a list of strings'
+
+        # if label and color are lists, check if they have the same length as param_to_plot
+        if len(label) != len(param_to_plot) or (color != None and len(color) != len(param_to_plot)):
+            raise ValueError(f'`param_to_plot`, `label` and `color` must have the same length, `param_to_plot` is a list of {len(param_to_plot)} elements')
+
+        #check if linestyle is a string, if not convert to list
```

Comment:
I think the following code converts linestyle into a list, even if it is a string.

Reply:
yes! it is intentional.
For each line, there will be a linestyle (string) in the list, which means that we need `len(list) = num lines`. 
If the input is a string, the list will be a list of n elements, and each element will be equal to the input